### PR TITLE
Added PHP rector to test process.

### PR DESCRIPTION
**Pull request by markdorison**
## Description
Added PHP rector to test process.

## Motivation / Context
Automate more things.

## Testing Instructions / How This Has Been Tested
Tests should fail, and then pass on subsequent commit.

### CODE REVIEW

File: src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php [hunk 1]
```diff
@@ -285,9 +285,9 @@ protected function landoUri(string $siteDir): string
             // proxy URL.
             $siteDomains = array_filter($landoCfg['proxy']['appserver'], fn($domain) =>
                 strpos($domain, $siteDir) === 0);
-            if (count($siteDomains) > 1) {
+            if (count((array) $siteDomains) > 1) {
```
Comment (markdorison) [inline src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php:288]: https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#countonnullrector

File: src/Robo/Plugin/Commands/DeployCommands.php [hunk 2]
```diff
@@ -85,7 +85,7 @@ public function deployDrupal(
             // https://github.com/drush-ops/drush/issues/2449#issuecomment-708655673
             ->exec("$appDirPath/vendor/bin/drush config:import --yes")
             ->run();
-        list('notify-slack' => $notifySlack, 'notify-slack-force' => $notifySlackForce) = $options;
+        ['notify-slack' => $notifySlack, 'notify-slack-force' => $notifySlackForce] = $options;
```
Comment (markdorison) [inline src/Robo/Plugin/Commands/DeployCommands.php:88]: https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#listtoarraydestructrector

File: src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php [hunk 3]
```diff
@@ -315,7 +315,7 @@ protected function landoUri(string $siteDir): string
     protected function drushDeployLando(string $siteDir = 'default'): Result
     {
         $this->io()->section('drush deploy.');
-        if (!class_exists('Drush\Commands\core\DeployCommands')) {
+        if (!class_exists(\Drush\Commands\core\DeployCommands::class)) {
```
Comment (markdorison) [inline src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php:318]: https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#stringclassnametoclassconstantrector

File: src/Robo/Plugin/Commands/DrupalStatusReportCommands.php [hunk 4]
```diff
@@ -84,7 +84,7 @@ public function drupalStatusReport(
                 ->printOutput(false)
                 ->run();
             $drushOutput = trim($result->getOutputData());
-            $reportJson = json_decode($drushOutput);
+            $reportJson = json_decode($drushOutput, null, 512, JSON_THROW_ON_ERROR);
```
Comment (markdorison) [inline src/Robo/Plugin/Commands/DrupalStatusReportCommands.php:87]: https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#jsonthrowonerrorrector

File: rector.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+use Rector\Config\RectorConfig;
+use Rector\Set\ValueObject\LevelSetList;
+
+return static function (RectorConfig $rectorConfig): void {
+    $rectorConfig->paths([
+        __DIR__ . '/src',
+        __DIR__ . '/tests',
+    ]);
+
+    // define sets of rules
```
Comment (adamzimmermann) [inline rector.php:14]: Indentation?

**Review by adamzimmermann — APPROVED** [review]
👍🏼

File: rector.php (hunk 5 above)
Comment (markdorison) [reply rector.php:14]: Usher follows [PSR-12](https://www.php-fig.org/psr/psr-12/). 4 spaces.

Comment (adamzimmermann) [reply rector.php:14]: Right, but this should still be aligned with line of code beneath it, right?

Comment (markdorison) [reply rector.php:14]: Ahh, it's actually the code below that is indented incorrectly. I will take care of it. Thanks for catching!

**Pull request merged**